src/dune/merlin_server.ml

```diff
@@ -0,0 +1,128 @@
+open! Stdune
+open Import
+
+module Dot = struct
+  type directive =
+    | EXCLUDE_QUERY_DIR
+    | TAG of string * string
+
+  type t = directive list
+
+  let error_tag = "ERROR"
+
+  (* [parse_line l] parses dune-generated .merlin files which only contain a
+     subset of merlin configuration options *)
+  let parse_line line =
```

Comment:
So I'm thinking, our plan is to add a version constraint between merlin and dune so that the new dune can only be used with a recent merlin that support this protocol. This means that we don't need to produce `.merlin` files in the source tree anymore.

What about generating a `.merlin.csexp` file instead to avoid the serialisation through the textual .merlin format?

Reply:
It sounds like a good idea.

If the new plan is to first release a version of dune with the `ocaml-merlin` command but keeping the `.merlin` files, should I:
1. Delay that good idea for when we do get rid of the `.merlin` files
2. Have both file on the build folder, `.merlin` and `.merlin.csexp`, but promote only the `.merlin`

Reply:
Let's delay it